### DISCUSSION TITLE
Keycloak in firefox is asking for refresh almost everytime to load a page

### DISCUSSION THREAD

**Discussion by sujaykulkarn** [keycloak/keycloak#33184] [Q&A]
Hi Community,

We are using Keycloak - 23.0.7 release. Deployed keycloak behind a reverse proxy. Everytime we try accessing console, it's asking for page refresh on each click in the UI. This is only happening in firefox, not with chrome. Chrome is working fine. Any suggestions or ideas on this behavior as it's blocking our testing on firefox. Please refer the attached images on the behavior. 

In logs also we are not seeing any error as such, so can you please help on this issue and how can we debug or fix this. Issue is observed when keycloak is deployed with more than one replica. 

KC version - 23.0.7
Firefox version used - latest
![image (1)](https://github.com/user-attachments/assets/b9aa4039-77b4-479c-8401-bb65611a9508)
![image (2)](https://github.com/user-attachments/assets/6b1f2d71-8e81-4540-ac06-d6e65968d379)

**Comment by jonkoops** [comment]
There have been several fixes landed in this regard since the release you are running. I'd recommend you upgrade to the latest version of Keycloak, it might just solve this specific issue.

**Reply by sujaykulkarn** [reply]
Hi @jonkoops ,
If you can share me the fixes details will be helpful for me.

**Reply by jonkoops** [reply]
I don't have time to look it up.

**Reply by sujaykulkarn** [reply]
@jonkoops,
Any tips if you remember on how we can solve this issue as work around, like any settings in browser or in LB? Because same configuration works well with chrome, only with firefox this issue, and our customer want's firefox support mainly.

**Reply by sujaykulkarn** [reply]
One more observation, this behavior is happening only when Keycloak is deployed with multiple replicas. So from caching perspective anything might have caused this?

**Reply by jonkoops** [reply]
Please try the latest version of Keycloak, if the behaviour is reproducible there you can submit this as a bug report and we will take a look at it. Older versions of Keycloak are not supported and are vulnerable to several critical security issues.

**Reply by Xulunix** [reply]
I can confirm that this error occurs from time to time on 25.0.6.
I'm using a 3-Instance Setup with a simple DNS Record pointing to all 3 IPs.

**Reply by sujaykulkarn** [reply]
Thanks @Xulunix for confirming. @jonkoops Since it's reproduced in latest aswell, will create it as issue.

**Reply by sujaykulkarn** [reply]
Created below issue for fixing this - https://github.com/keycloak/keycloak/issues/33263

**Reply by jonkoops** [reply]
Thanks, let's see if we can reproduce it there.

**Comment by ShaswataJash** [comment]
Is there any possibility that due to 'pre-fetching' functionality of Chrome, it is somehow refreshing the page internally and that is why we are not seeing the problem in Chrome?
